.github/workflows/docker.yml

```diff
@@ -14,6 +12,52 @@ jobs:
       -
         name: Checkout
         uses: actions/checkout@v3
+      - 
+        name: Check whether there are any commits since this run was last triggered and push them and/or set the output
+        id: should_run
+        if: ${{ (github.event_name == 'workflow_dispatch' && contains(github.event.inputs.message, '[main]')) }}
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+
+          # Defensive: update submodules in the main branch
+          git checkout main
+          git submodule update --init --recursive --remote --rebase --force
+          git add .
+          git commit -am "Update submodules [skip ci]" --no-verify || true
+          git push origin main --no-verify
+          git checkout -
+
+          git submodule update --init --recursive --remote --rebase --force
+
+          CHANGES_COUNT=$(git diff --shortstat origin/main | wc -l)
+          MANUAL_REBUILD="${{ github.event_name == 'workflow_dispatch' }}"
+          VERSION_BUMP="${{ contains(github.event.inputs.message, '[version bump]') }}"
+          if [ $CHANGES_COUNT -gt 0 ] || [[ $MANUAL_REBUILD == "true" && $VERSION_BUMP == "true" ]]; then
+            # Do the version bump in the 'main' branch ONLY if
+            #   there were other changes coming from the 'main' branch (or)
+            #   this is a manual trigger with the key-phrase
+            git checkout main
+            TAG_NAME=$(npm version -m "%s [skip ci]" patch)
+            git commit --all --amend --no-edit --no-verify
+            git push origin main --no-verify
+            git tag -f $TAG_NAME
+            git push origin --tags --no-verify
+
+            git checkout main
```

Comment:
Is this line required? (since you are already doing a checkout in line 40)?

Reply:
it shouldn't be, I'll remove it, thanks!